disnake/member.py

```diff
@@ -650,6 +658,21 @@ def voice(self) -> Optional[VoiceState]:
         """Optional[:class:`VoiceState`]: Returns the member's current voice state."""
         return self.guild._voice_state_for(self._user.id)
 
+    @property
+    def current_timeout(self) -> Optional[datetime.datetime]:
+        """Optional[:class:`datetime.datetime`]: Returns the datetime when the time out expires, if any.
+
+        .. versionadded:: 2.3
+        """
+
+        if (
+            self._communication_disabled_until
+            and self._communication_disabled_until < utils.utcnow()
+        ):
+            return None
```

Comment:
```suggestion
        if self._communication_disabled_until is None:
            return None
        if self._communication_disabled_until < utils.utcnow():
            self._communication_disabled_until = None
            return None
```

Might be a nice idea to clear the timeout?